pkg/controller/promotionreconciler/reconciler.go

```diff
@@ -239,3 +260,59 @@ func configIndexFn(in cioperatorapi.ReleaseBuildConfiguration) []string {
 func configIndexKeyForIST(ist *imagev1.ImageStreamTag) string {
 	return ist.Namespace + "/" + ist.Name
 }
+
+func handleCIOpConfigChange(registryClient ctrlruntimeclient.Client,
+	ciOperatorConfigGetter ciOperatorConfigGetter,
+	prowJobEnqueuer prowjobreconciler.Enqueuer,
+	githubClient githubClient,
+	delta agents.IndexDelta,
+	log *logrus.Entry) error {
+	log = log.WithField("indexKey", delta.IndexKey)
+	log.Debug("Handling CIOpConfig change")
+	// We only care about new additions
+	if len(delta.Added) == 0 {
+		return nil
+	}
+	slashSplit := strings.Split(delta.IndexKey, "/")
+	if len(slashSplit) != 2 {
+		return fmt.Errorf("got an index delta event with a key that is not a valid namespace/name identifier: %s", delta.IndexKey)
+	}
+	namespace, name := slashSplit[0], slashSplit[1]
+	var ist imagev1.ImageStreamTag
+	if err := registryClient.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: name}, &ist); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get imagestreamtag %s/%s: %w", namespace, name, err)
+		}
+		ist = imagev1.ImageStreamTag{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		}
```

Comment:
Can't we just enqueue this object and handle `IsNotFound` in `reconcile` instead of performing essentially the same actions here?

Reply:
The PR handles ONLY `IsNotFound` cases, i.e., ci-op's defines a promotion tag but we cannot find it on the cluster. Then we trigger a image job for it.

Reply:
I guess I got your point now.
Enqueuing an object that does not exist feel odd: How do we tell if the tag is deleted or it has not been created yet when it is reconciled.